bonfire/bonfire.py

```diff
@@ -953,14 +954,22 @@ def _get_apps_config(
     return apps_config
 
 
-def _get_env_name(target_namespace, env_name):
+def _get_env_name(ns, env_name):
     if not env_name:
-        if not target_namespace:
-            _error(
-                "unable to infer name of ClowdEnvironment if namespace not provided."
-                "  Please run with one of: --clowd-env or --namespace"
-            )
-        env_name = conf.ENV_NAME_FORMAT.format(namespace=target_namespace)
+        if ns:
+            log.info("searching for ClowdEnvironment tied to ns '%s'...", ns)
+            match = find_clowd_env_for_ns(ns)
+            if not match:
+                log.warning(
+                    "could not find a ClowdEnvironment with target ns '%s'.  "
+                    "Specify one with '--clowd-env' if needed."
+                )
+            else:
+                env_name = match["metadata"]["name"]
+        else:
+            log.warning("neither '--clowd-env' nor '--namespace' provided")
+
+    log.info("templates will be processed with parameter ENV_NAME='%s'", env_name)
```

Comment:
This is good, decomposing this code out into a method to be used elsewhere. But, could I propose expressing this differently?

```
def _log_get_env_name(env_name):
    log.info("templates will be processed with parameter ENV_NAME='%s'", env_name)
    return env_name

def _get_env_name(ns, env_name):
    if env_name:
        return _log_get_env_name(env_name)

    if not ns:
        log.warning("neither '--clowd-env' nor '--namespace' provided")
        return _log_get_env_name(None)
    
    log.info("searching for ClowdEnvironment tied to ns '%s'...", ns)
    match = find_clowd_env_for_ns(ns)
    if not match:
        log.warning(
            "could not find a ClowdEnvironment with target ns '%s'.  "
            "Specify one with '--clowd-env' if needed."
        )
        return _log_get_env_name(None)

    return _log_get_env_name(match["metadata"]["name"])
```

The suggestion isn't merely stylistic, or that I think its better aesthetically. When I reviewed this method I had to trace through each one of the nested ifs to try and understand every path we could take through the method. I found it hard to do, and I know if I had to work on this method later I'd have to spend a while figuring out what all are the paths through here and what's being set where. The proposed version is flat, has no nested logic, and we always return early. I add the log method so we still get the DRY log message you were going for without needing to let all execution paths dump into the return.

Reply:
I like it, clearer logic.